pkg/server/chaosd/disk.go

```diff
@@ -38,84 +37,96 @@ const DDWritePayloadCommand = "dd if=/dev/zero of=%s bs=%s count=%s oflag=dsync"
 const DDReadPayloadCommand = "dd if=%s of=/dev/null bs=%s count=%s iflag=dsync,direct,fullblock"
 
 func (disk diskAttack) Attack(options core.AttackConfig, env Environment) (err error) {
-	attack := options.(*core.DiskCommand)
+	attack := options.(*core.DiskOption)
 
 	if options.String() == core.DiskFillAction {
 		return disk.attackDiskFill(attack)
 	}
 	return disk.attackDiskPayload(attack)
 }
 
-func (diskAttack) attackDiskPayload(payload *core.DiskCommand) error {
+func (diskAttack) attackDiskPayload(payload *core.DiskOption) error {
+	var cmdFormat string
 	switch payload.Action {
 	case core.DiskWritePayloadAction:
+		cmdFormat = DDWritePayloadCommand
 		if payload.Path == "" {
-			payload.Path = "/dev/null"
-		}
-		cmd := exec.Command("bash", "-c", fmt.Sprintf(DDWritePayloadCommand, payload.Path, "1M", payload.Size))
-		output, err := cmd.CombinedOutput()
-
-		if err != nil {
-			log.Error(string(output), zap.Error(err))
-		} else {
-			log.Info(string(output))
+			var err error
+			payload.Path, err = utils.CreateTempFile()
+			if err != nil {
+				log.Error(fmt.Sprintf("unexpected err when CreateTempFile in action: %s", payload.Action))
+				return err
+			}
+			defer func() {
+				err := os.Remove(payload.Path)
+				if err != nil {
+					log.Error(fmt.Sprintf("unexpected err when removing temp file %s", payload.Path), zap.Error(err))
+				}
+			}()
 		}
-		return err
 	case core.DiskReadPayloadAction:
+		cmdFormat = DDReadPayloadCommand
 		if payload.Path == "" {
-			err := errors.Errorf("empty read payload path")
-			log.Error(fmt.Sprintf("payload action: %s", payload.Action), zap.Error(err))
-			return err
+			path, err := utils.GetRootDevice()
+			if err != nil {
+				log.Error("err when GetRootDevice in reading payload", zap.Error(err))
+				return err
+			}
+			if path == "" {
+				err = fmt.Errorf("can not get root device path")
```

Comment:
here use `fmt.Errorf`, but L82 use `errors.Errorf`, it's better to unify them, only use one

Reply:
OK，I aggree with you.